docs/Developers/IO.rst

```diff
@@ -0,0 +1,35 @@
+coala's IO Mechanism
+====================
+
+The interface
+-------------
+
+Instead of directly loading files from a project directory coala uses a
+class called ``FileFactory`` that provides an interface for dealing with
+files.
+
+``FileFactory`` provides the following advantages as compared to simple file
+loading.
+
+- Lazy-loading of files: Files don't have to be immediately loaded into the
+  memory, instead coala collects all the ``FileFactory`` objects first which
+  are in turn used to access the file contents when needed.
+- Various modes to access contents: ``FileFactory`` provides three modes to
+  access the file contents. As a string using ``FileFactory.string``, as a
+  list using ``FileFactory.lines`` and in raw format using ``FileFactory.raw``.
+  To access a particular line in the file one can use
+  ``FileFactory.line(line_number)``.
```

Comment:
For `raw` I would definitely mention the idea to deploy analysis that works on raw files only. As long as you don't use `string`, the contents will be never decoded, and you never run into problems. No quirky `USE_RAW_FILES` any more.